### PR TITLE
Fix cybernetic communications implants

### PR DESCRIPTION
**Pull request by HammerGS**
Summary

  Implements the Cybernetic Communications Implant (MD_COMM_IMPLANT) bonus for artillery spotting and fixes the display of
  spotter bonuses.

  What Changed
  - Comm Implant now provides -1 to artillery attacks when the spotter has the implant (non-infantry only)
  - Forward Observer and Comm Implant bonuses now display as separate line items instead of a combined "Spotting modifier"
  - Spotter bonuses only apply on adjusted fire (after first shell lands), per Tactical Operations rules

  Testing Notes for QA

  Important: Artillery vs LRM Spotting
  - LRM spotting is EXPLICIT: Player designates a specific unit to spot, that unit takes an action
  - Artillery spotting is IMPLICIT: Any friendly unit with LOS to the target hex automatically spots - no player action
  required

 Test Cases
  1. First shot at a hex: Should show +0 (spotter available - bonus applies after first shell lands) - no bonuses applied yet
  2. After first shell lands: Subsequent shots should show separate modifiers:
    - -1 (spotter is forward observer) if spotter has FO SPA
    - -1 (spotter has comm implant) if spotter has Comm Implant
  3. Infantry spotters: Should NOT get comm implant bonus (even if they somehow have it)
  4. Spotter must have LOS to target hex

  Files Modified

  - messages.properties (client) - Added message keys for spotter display text
  - messages.properties (options) - Added Comm Implant option description
  - ArtilleryTracker.java - Added spotterHasCommImplant and spotterHasForwardObs flags to persist spotter abilities across
  turns
  - ComputeToHit.java - Updated artillery to-hit preview to show informational message on first shots and display FO/Comm
  Implant bonuses separately
  - PilotOptions.java - Added Comm Implant to pilot options
  - ArtilleryHandlerHelper.java - Refactored spotter selection logic for better readability, breaking out filter conditions
  into named variables. Added debug logging to trace which entities are evaluated as potential spotters and why they pass or
  fail validation.
  - ArtilleryBayWeaponIndirectFireHandler.java - Added comm implant bonus support for artillery bays
  - ArtilleryWeaponIndirectFireHandler.java - Added isAdjustedFire check to ensure spotter bonuses only apply after first
  shell lands, refactored bonus display to show FO and Comm Implant separately, added debug logging
  - TWGameManager.java - Added Comm Implant bonus for infantry minefield avoidance (-1 to detonation roll). Added debug
  logging for artillery spotter declaration to help troubleshoot spotter selection issues.
  - ArtilleryTrackerTest.java - Added unit tests for comm implant/FO flag getters/setters
  - ArtilleryHandlerHelperTest.java - New unit tests for findSpotter() edge cases

### CODE REVIEW

File: megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java [hunk 1]
```diff
@@ -183,17 +187,52 @@ public boolean handle(GamePhase phase, Vector<Report> vPhaseReport) {
...
+                if (gunneryMod != 0) {
+                    toHit.addModifier(gunneryMod, Messages.getString("WeaponAttackAction.SpotterGunnery"));
+                }
+
+                // Add Forward Observer modifier separately
+                boolean hasFO = bestSpotter.get().hasAbility(OptionsConstants.MISC_FORWARD_OBSERVER);
+                logger.debug("  Spotter FO check: hasFO={}, modifier={}", hasFO, hasFO ? -2 : 0);
+                if (hasFO) {
+                    toHit.addModifier(-2, Messages.getString("WeaponAttackAction.SpotterFO"));
+                }
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java:223]: The -2 is correct.

File: megamek/src/megamek/common/actions/compute/ComputeToHit.java [hunk 2]
```diff
@@ -70,12 +70,11 @@
 import megamek.common.weapons.battleArmor.clan.CLBALBX;
 import megamek.common.weapons.bayWeapons.ScreenLauncherBayWeapon;
 import megamek.common.weapons.capitalWeapons.CapitalMissileWeapon;
+import megamek.common.weapons.handlers.ARADEquipmentDetector;
 import megamek.common.weapons.lasers.VariableSpeedPulseLaserWeapon;
 import megamek.common.weapons.lasers.innerSphere.ISBombastLaser;
 import megamek.common.weapons.lrms.LRTWeapon;
 import megamek.common.weapons.srms.SRTWeapon;
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/actions/compute/ComputeToHit.java:77]: No change needed.

File: megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java [hunk 3]
```diff
@@ -183,17 +186,48 @@ public boolean handle(GamePhase phase, Vector<Report> vPhaseReport) {
...
+
+                // Comm implant bonus only applies to non-infantry spotters
+                boolean isInfantry = bestSpotter.get() instanceof Infantry;
+                boolean hasCommImplant = bestSpotter.get().hasAbility(OptionsConstants.MD_COMM_IMPLANT);
+                logger.debug("  Spotter comm implant check: isInfantry={}, hasCommImplant={}",
+                      isInfantry,
+                      hasCommImplant);
+                if (!isInfantry && hasCommImplant) {
+                    toHit.addModifier(-1, "spotter has comm implant");
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java]: Fixed

File: megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java [hunk 4]
```diff
@@ -183,17 +186,48 @@ public boolean handle(GamePhase phase, Vector<Report> vPhaseReport) {
...
-                toHit.addModifier(mod, "Spotting modifier");
+                if (gunneryMod != 0) {
+                    toHit.addModifier(gunneryMod, "spotter gunnery modifier");
+                }
+
+                // Add Forward Observer modifier separately
+                boolean hasFO = bestSpotter.get().hasAbility(OptionsConstants.MISC_FORWARD_OBSERVER);
+                logger.debug("  Spotter FO check: hasFO={}", hasFO);
+                if (hasFO) {
+                    toHit.addModifier(-1, "spotter is forward observer");
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java]: Fixed

File: megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryBayWeaponIndirectFireHandler.java [hunk 5]
```diff
@@ -264,9 +265,20 @@ public boolean accept(Entity entity) {
...
+                commImplantMod = -1;
+            }
             int mod = (bestSpotter.getCrew().getGunnery() - 4) / 2;
-            mod += foMod;
+            mod += foMod + commImplantMod;
             toHit.addModifier(mod, "Spotting modifier");
+            // Comm implant bonus applies to non-infantry spotters for artillery
+            if (!(bestSpotter instanceof Infantry) &&
+                  bestSpotter.hasAbility(OptionsConstants.MD_COMM_IMPLANT)) {
+                toHit.addModifier(-1, "Spotter has comm implant");
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryBayWeaponIndirectFireHandler.java]: Fixed

File: megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryBayWeaponIndirectFireHandler.java [hunk 6]
```diff
@@ -264,9 +265,20 @@ public boolean accept(Entity entity) {
...
+            }
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryBayWeaponIndirectFireHandler.java]: Fixed

File: megamek/unittests/megamek/common/weapons/ArtilleryHandlerHelperTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+     * Test that isForwardObserver returns false for entities without FO ability.
+     */
+    @Test
+    void isForwardObserver_withoutFOAbility_returnsFalse() {
+        Entity entity = mock(Entity.class);
+        when(entity.hasAbility(OptionsConstants.MISC_FORWARD_OBSERVER)).thenReturn(false);
+
+        assertFalse(ArtilleryHandlerHelper.isForwardObserver(entity));
+    }
+}
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/unittests/megamek/common/weapons/ArtilleryHandlerHelperTest.java:185]: Fixed

File: megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java [hunk 8]
```diff
@@ -183,17 +186,48 @@ public boolean handle(GamePhase phase, Vector<Report> vPhaseReport) {
...
+                toHit.removeModifier("comm implant");
+
+                // Add spotter gunnery modifier
+                int gunneryMod = ((useArtillerySkill ?
                       bestSpotter.get().getCrew().getArtillery() :
                       bestSpotter.get().getCrew().getGunnery()) - 4) / 2;
-                mod += foMod;
-                toHit.addModifier(mod, "Spotting modifier");
+                if (gunneryMod != 0) {
+                    toHit.addModifier(gunneryMod, "spotter gunnery modifier");
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/weapons/handlers/artillery/ArtilleryWeaponIndirectFireHandler.java]: This has been fixed

File: megamek/src/megamek/server/totalWarfare/TWGameManager.java [hunk 9]
```diff
@@ -7501,6 +7501,11 @@ private boolean enterMinefield(Entity entity, Coords c, int curElev, boolean isO
                 if (entity.hasAbility(OptionsConstants.MISC_EAGLE_EYES)) {
                     target += 2;
                 }
+                // Comm implant makes it easier for infantry to avoid mines
+                if ((entity instanceof Infantry) &&
+                      entity.hasAbility(OptionsConstants.MD_COMM_IMPLANT)) {
+                    target -= 1;
```
Comment (Sleet01) [inline megamek/src/megamek/server/totalWarfare/TWGameManager.java]: Believe this should be `target += 1;` as a higher target is easier to come in under; see Eagle Eyes section directly above.

Edit: see line 7529; the entity misses the minefield if they roll _below_ the target.

**Review by Sleet01 — CHANGES_REQUESTED** [review]
Line 7507 needs its polarity swapped, otherwise looks good.

File: megamek/src/megamek/server/totalWarfare/TWGameManager.java (hunk 9 above)
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWGameManager.java]: Fixed with https://github.com/MegaMek/megamek/pull/7700/commits/feb8d5426d8abd3ea8aa1ce4e2eb98bc604c2e0e
